src/main/java/cs2103/v15_1j/jimjim/view/TaskViewController.java

```diff
@@ -0,0 +1,345 @@
+        taskCompletedColumn.setPrefWidth(TASK_COMPLETED_COLUMN_WIDTH);
+        taskCompletedColumn.setResizable(false);
+        
+		return taskCompletedColumn;
+	}
+	
+	private TableColumn<Task, LocalDateTime> setUpTaskDateTimeColumn(){
+		TableColumn<Task, LocalDateTime> taskDateColumn = new TableColumn<Task, LocalDateTime>();
+	    
+        
+        DateTimeFormatter myDateFormatter = DateTimeFormatter.ofPattern("dd.MM hh:mm");
+        taskDateColumn.setCellValueFactory(cellData -> cellData.getValue().dateTimeProperty());
+        taskDateColumn.setCellFactory(column -> {
+	         return new TableCell<Task, LocalDateTime>() {
+	             @Override
+	             protected void updateItem(LocalDateTime item, boolean empty) {
+	                 super.updateItem(item, empty);
+	
+	                 if (item == null || empty) {
+	                     setText(null);
+	                     setStyle("");
+	                 } else {
+	                     setText(myDateFormatter.format(item));
+	                 }
+	             }
+	         };
+	     });
+        
+        return taskDateColumn;
+	}
+	
+	@SuppressWarnings("unchecked")
+	private TableView<Task> setUpTaskTable(TableColumn<Task, Boolean> taskCompletedColumn, TableColumn<Task, String>
+		taskNameColumn, TableColumn<Task, LocalDateTime> taskDateTimeColumn){
+		TableView<Task> taskTable = new TableView<Task>();
+		taskTable.setColumnResizePolicy(TableView.CONSTRAINED_RESIZE_POLICY);
+		taskTable.getColumns().addAll(taskCompletedColumn, taskNameColumn, taskDateTimeColumn);
+		taskTable.setEditable(true);
+		taskTable.setItems(taskData);
+        
+		taskTable.setRowFactory( tv -> {
+            TableRow<Task> row = new TableRow<>();
+            row.setOnMouseClicked(event -> {
+                if (event.getClickCount() == 1 && (! row.isEmpty()) ) {
+                    Task rowData = row.getItem();
+                    taskPane.setDetailNode(setUpDetailNode(rowData));
+                    taskPane.setShowDetailNode(true);
+                }
+            });
+            return row ;
+        });
+		
+		return taskTable;
+	}
+	
+	public void refreshUI(List<TaskEvent> tempList){
+    	taskData.clear();
+    	for(TaskEvent te: tempList){
+    		taskData.add((Task) te);
+    	}
+    }
+	
+	public void focusCommandBar(){
+    	commandBar.requestFocus();
+    }
+	
+    private void handleCommand() {
+    	if (commandBar.getText() != null) {
+    		String status = uiController.executeCommand(commandBar.getText());
+    		displayMessage(status);
+            commandBar.setText("");
+
+            
+        } else {
+            // Nothing entered.
+            Alert alert = new Alert(AlertType.WARNING);
+            alert.initOwner(uiController.getPrimaryStage());
+            alert.setTitle("Empty Command");
+            alert.setHeaderText("No Command Entered");
+            alert.setContentText("Please enter a command.");
+
```

Comment:
how about don't do anything? What's the point of alerting that nothing has been entered